lib/farscape/agent.rb

```diff
@@ -1,16 +1,31 @@
 require 'farscape/representor'
 require 'farscape/clients'
+require 'farscape/discovery'
 
 module Farscape
   class Agent
-    
+
     include BaseAgent
-    
+
     PROTOCOL = :http
 
     attr_reader :media_type
     attr_reader :entry_point
-    
+
+    class << self
+      # Prevents multiple threads to access the same agent.
```

Comment:
not a native speaker but IMHO it should be "... threads **from accessing** the same agent"